packages/angular/projects/clr-angular/src/data/datagrid/datagrid-row.spec.ts

```diff
@@ -121,6 +121,40 @@ export default function (): void {
       });
     });
 
+    describe('Conditional selection switch order or arguments', function () {
+      let context: TestContext<ClrDatagridRow, SelectableRowOrder>;
+      let selectionProvider: Selection;
+      let checkbox: HTMLElement;
+
+      beforeEach(function () {
+        context = this.create(ClrDatagridRow, SelectableRowOrder, DATAGRID_SPEC_PROVIDERS);
+        selectionProvider = TestBed.get(Selection);
+        TestBed.get(Items).all = [{ id: 1 }, { id: 2 }];
+      });
+
+      it('should toggle when clrDgSelectable is false for type  SelectionType.Multi', () => {
+        selectionProvider.selectionType = SelectionType.Multi;
+        context.testComponent.clrDgSelectable = false;
+        context.detectChanges();
+        checkbox = context.clarityElement.querySelector("input[type='checkbox']");
+
+        expect(checkbox.getAttribute('disabled')).toBe('true');
+        expect(checkbox.getAttribute('aria-disabled')).toBe('true');
+
+        context.clarityDirective.toggle();
+        expect(context.clarityDirective.selected).toBe(true);
```

Comment:
Two things here:

1. The description looks off. When `clrDgSelectable` is false, you **shouldn't be able to toggle** through the selection checkbox's UI because it's disabled. I see you've tested whether it's getting `disabled` attributes.

2. The lines I've highlighted here doesn't belong here. These lines are testing the API part. And It looks like you are testing it in the test right below it.